workspaces/arborist/lib/override-set.js

```diff
@@ -142,6 +181,24 @@ class OverrideSet {
 
     return ruleset
   }
+
+  static findSpecificOverrideSet (first, second) {
```

Comment:
@jdalton I moved the functions here, because fundamentally they are about comparing override sets, so this is the most natural location for them I think.

Reply:
Is there anything else, or do you approve the PR?

Reply:
@AlonNavon 
I think it looks great. Could you add code comments inline explaining the cases when the .silly is called (the one you changed from throwing to .silly). This will help folks if they see the logging know why. 

Then the only other thing is adding any unit tests to cover some of the things you discovered and tweaked.

Reply:
Hey @wraithgar, can you run the tests? To make sure nothing got accidentally broken with all the changes.

Reply:
@AlonNavon Any insight into the issue above?